src/widgets/index/__tests__/index-test.ts

```diff
@@ -1508,6 +1510,88 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index/js/"
         expect(level1.getWidgetState).toHaveBeenCalledTimes(0);
       });
 
+      it('updates the local `uiState` when they differ on first render', () => {
+        const instance = index({ indexName: 'indexName' });
+
+        instance.addWidgets([createSearchBox()]);
+
+        instance.init(createInitOptions());
+
+        expect(instance.getWidgetState({})).toEqual({
+          indexName: {},
+        });
+
+        // Simulate a change that does not emit (like `searchFunction`)
+        instance.getHelper()!.overrideStateWithoutTriggeringChangeEvent({
+          ...instance.getHelper()!.state,
+          query: 'Apple iPhone',
+        });
+
+        instance.render(createRenderOptions());
+
+        // @TODO: assert that onChange is called once
+        // expect(...onChange).toHaveBeenCalledTimes(1);
+        expect(instance.getWidgetState({})).toEqual({
+          indexName: {
+            query: 'Apple iPhone',
+          },
+        });
+
+        // Simulate a change that does not emit (like `searchFunction`)
+        instance.getHelper()!.overrideStateWithoutTriggeringChangeEvent({
+          ...instance.getHelper()!.state,
+          query: 'Apple iPhone XS',
+        });
+
+        instance.render(createRenderOptions());
+
+        // @TODO: assert that onChange is called once
```

Comment:
```suggestion
        // @TODO: assert that onChange is called second time
```

Reply:
Actually no it has to be called only once. We only update the `uiState` on the first render.